### ISSUE TITLE
Missing null deref warning on nullable enabled code

### ISSUE THREAD

**Issue by ryzngard** [dotnet/roslyn#52352]
**Steps to Reproduce**:

Simplified code from Roslyn that I was working on: 

```c#
class C
{
  async Task M()
  {
      ...
      var operation = semanticModel.GetOperation(node, cancellationToken);
      await operationCollector.VisitAsync(operation, cancellationToken).ConfigureAwait(false);
      ...
  }

class OperationCollector
{
    Task VisitAsync(IOperation operation, CancellationToken cancellationToken) => Task.CompletedTask;
}
```

**Expected Behavior**: `await operationCollector.VisitAsync(operation, cancellationToken).ConfigureAwait(false);` should report possible null deref, GetOperation returns `IOperation?`

**Actual Behavior**: No warning

**Version Used**: 
Microsoft Visual Studio Enterprise 2019 Int Preview
Version 16.10.0 Preview 2.0 [31130.22.main]
VisualStudio.16.IntPreview/16.10.0-pre.2.0+31130.22.main
Microsoft .NET Framework
Version 4.8.04084

Installed Version: Enterprise

Visual C++ 2019   00435-60000-00000-AA483
Microsoft Visual C++ 2019

ASP.NET and Web Tools 2019   16.10.306.5925
ASP.NET and Web Tools 2019

ASP.NET Core Razor Language Services   16.1.0.2117202+fdd84012f69ea6eb239477cb392aaf4e67247894
Provides languages services for ASP.NET Core Razor.

ASP.NET Web Frameworks and Tools 2019   16.10.306.5925
For additional information, visit https://www.asp.net/

Azure App Service Tools v3.0.0   16.10.306.5925
Azure App Service Tools v3.0.0

Azure Functions and Web Jobs Tools   16.10.306.5925
Azure Functions and Web Jobs Tools

C# Tools   3.10.0-2.21179.6+07d62e07f318a9081fd62277342fd2d8c11e7ba2
C# components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Common Azure Tools   1.10
Provides common services for use by Azure Mobile Services and Microsoft Azure Tools.

IntelliCode Extension   1.0
IntelliCode Visual Studio Extension Detailed Info

Microsoft Azure Tools   2.9
Microsoft Azure Tools for Microsoft Visual Studio 2019 - v2.9.40309.1

Microsoft Continuous Delivery Tools for Visual Studio   0.4
Simplifying the configuration of Azure DevOps pipelines from within the Visual Studio IDE.

Microsoft JVM Debugger   1.0
Provides support for connecting the Visual Studio debugger to JDWP compatible Java Virtual Machines

Microsoft Library Manager   2.1.113+g422d40002e.RR
Install client-side libraries easily to any web project

Microsoft MI-Based Debugger   1.0
Provides support for connecting Visual Studio to MI compatible debuggers

Microsoft Visual C++ Wizards   1.0
Microsoft Visual C++ Wizards

Microsoft Visual Studio Tools for Containers   1.2
Develop, run, validate your ASP.NET Core applications in the target environment. F5 your application directly into a container with debugging, or CTRL + F5 to edit & refresh your app without having to rebuild the container.

Microsoft Visual Studio VC Package   1.0
Microsoft Visual Studio VC Package

NuGet Package Manager   5.10.0
NuGet Package Manager in Visual Studio. For more information about NuGet, visit https://docs.nuget.org/

ProjectServicesPackage Extension   1.0
ProjectServicesPackage Visual Studio Extension Detailed Info

Snapshot Debugging Extension   1.0
Snapshot Debugging Visual Studio Extension Detailed Info

SQL Server Data Tools   16.0.62103.10080
Microsoft SQL Server Data Tools

Syntax Visualizer   1.0
An extension for visualizing Roslyn SyntaxTrees.

Test Adapter for Boost.Test   1.0
Enables Visual Studio's testing tools with unit tests written for Boost.Test.  The use terms and Third Party Notices are available in the extension installation directory.

Test Adapter for Google Test   1.0
Enables Visual Studio's testing tools with unit tests written for Google Test.  The use terms and Third Party Notices are available in the extension installation directory.

TypeScript Tools   16.0.30324.2001
TypeScript Tools for Microsoft Visual Studio

Visual Basic Tools   3.10.0-2.21179.6+07d62e07f318a9081fd62277342fd2d8c11e7ba2
Visual Basic components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Visual F# Tools   16.10.0-beta.21171.1+437bc2042180d70599f4a73b55f00ac04c49ba8e
Microsoft Visual F# Tools

Visual Studio Code Debug Adapter Host Package   1.0
Interop layer for hosting Visual Studio Code debug adapters in Visual Studio

Visual Studio Container Tools Extensions   1.0
View, manage, and diagnose containers within Visual Studio.

Visual Studio Tools for CMake   1.0
Visual Studio Tools for CMake

Visual Studio Tools for Containers   1.0
Visual Studio Tools for Containers

**Comment by jaredpar** [comment]
I'm unable to reproduce this situation. Expanded out your sample code in sharplab and I see the warning as expected 

https://sharplab.io/#v2:EYLgtghglgdgNAExAagD4AEAMACdBGAOgBUALAJwFMIFYBzAbgFgAoLXPAVieYGIYBXADaCIwQRWwUYo8SxboATNgDCLAN4tsuABy4AbNgCyACgDKFSDAAuUAMaGA9ggqDsAZwsRrdx88FxsUwBPawgADwA5JwkYaICAeQAHCjIIGwcYZQdhClsrBzJsB2TU9Mzs8TyCgOUvWxcRMqIHAGspbFs6hrSoDOa2mABKTWwNZi0J7AA3CELilJ6M7ABed09ve2jBAgBxCiskhbLjWOcAzph64UWYfqlB7kmtdABOIpKbrJyqsgIANSgbigVgAgm4QrZjPNSr14B0utcmq17gQsjAAGZQWj8SgggDu0CsxnREEEHgeIwAvixqcx5EpDjCMl9KvkyOoRlpYFYUtJXOgDACgaDwZdjABJRk3d5HWE1BGNWF3GDwy7dJEDQYrAB8+lRDjAiXEPIQAu4tPpgXWNk2fg54wm3N5pOwko+ZQA/Ng9gd3bCzCErOEos5sKcKPK1YilciVRcroq+rGtctdVZyA48WGhIIAITmuSsJTBUKRaKjbAW5hOsgk+quqVlCuUoA
